packages/components/src/DataTable/DataTable.test.tsx

```diff
@@ -0,0 +1,185 @@
+import React from "react";
+import renderer from "react-test-renderer";
+import { DataTable } from ".";
+
+const data = [
+  {
+    name: "Eddard",
+    house: "Stark",
+    region: "North",
+    sigil: "Direwolf",
+    isAlive: "No",
+  },
+  {
+    name: "Catelyn",
+    house: "Stark",
+    region: "North",
+    sigil: "Direwolf",
+    isAlive: "No",
+  },
+  {
+    name: "Jon Snow",
+    house: "Stark",
+    region: "North",
+    sigil: "Direwolf",
+    isAlive: "Yes",
+  },
+  {
+    name: "Robert",
+    house: "Stark",
+    region: "North",
+    sigil: "Direwolf",
+    isAlive: "No",
+  },
+  {
+    name: "Rickon",
+    house: "Stark",
+    region: "North",
+    sigil: "Direwolf",
+    isAlive: "No",
+  },
+  {
+    name: "Robert",
+    house: "Baratheon",
+    region: "Stormlands",
+    sigil: "Black Stag",
+    isAlive: "No",
+  },
+  {
+    name: "Cercei",
+    house: "Lannister",
+    region: "Westerlands",
+    sigil: "Golden Lion",
+    isAlive: "Yes",
+  },
+  {
+    name: "Sansa",
+    house: "Stark",
+    region: "North",
+    sigil: "Direwolf",
+    isAlive: "Yes",
+  },
+  {
+    name: "Arya",
+    house: "Stark",
+    region: "North",
+    sigil: "Direwolf",
+    isAlive: "Yes",
+  },
+  {
+    name: "Bran",
+    house: "Stark",
+    region: "North",
+    sigil: "Direwolf",
+    isAlive: "Yes",
+  },
+  {
+    name: "Joffrey",
+    house: "Baratheon",
+    region: "Stormlands",
+    sigil: "Black Stag",
+    isAlive: "No",
+  },
+  {
+    name: "Myrcella",
+    house: "Baratheon",
+    region: "Stormlands",
+    sigil: "Black Stag",
+    isAlive: "Yes",
+  },
+  {
+    name: "Tommen",
+    house: "Baratheon",
+    region: "Stormlands",
+    sigil: "Black Stag",
+    isAlive: "Yes",
+  },
+];
+const columns = [
+  {
+    accessorKey: "name",
+    cell: info => info.getValue(),
+  },
+  {
+    accessorKey: "house",
+    cell: info => info.getValue(),
+  },
+  {
+    accessorKey: "region",
+    cell: info => info.getValue(),
+  },
+  {
+    accessorKey: "sigil",
+    cell: info => info.getValue(),
+  },
+  {
+    accessorKey: "isAlive",
+    cell: info => info.getValue(),
+  },
+];
+const setPagination = () => {
+  console.log("Change pagination");
+};
+const setSorting = () => {
+  console.log("Change sorting");
+};
+const pageIndex = 0;
+const pageSize = 10;
+const sorting = { manualSorting: false };
+const pagination = { manualPagination: false, itemsPerPage: [10, 20, 30] };
+const manualPagination = {
+  manualPagination: true,
+  onPaginationChange: setPagination,
+  pageCount: Math.ceil(100 / pageSize),
+  state: {
+    pageIndex,
+    pageSize,
+  },
+};
+const manualSorting = {
+  manualSorting: true,
+  state: [{ id: "name", desc: false }],
+  onSortingChange: setSorting,
+};
+
+it("renders a basic table table", () => {
+  const tree = renderer
+    .create(<DataTable data={data} columns={columns} />)
+    .toJSON();
+  expect(tree).toMatchSnapshot();
```

Comment:
Though Snapshot tests still exist, we are trying to move away from adding them given they are not reliable, and people often miss actual breaking changes when updating snapshots since they so easily change.

I'd prefer to see a way of testing the data you want to be displayed is getting displayed with checks like `getBy...`, `queryBy...`, `expect(myMockedPaginationChange).toBeCalled()` etc.

Reply:
Is this something you plan to address??

Reply:
@jsuazod

Reply:
Sure, we can add those tests, i agree 100%, snapshot tests are not reliable.

Reply:
✅